src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -389,16 +391,34 @@ protected boolean hasExceptionMessage(Object receiver,
     @CoreMethod(names = "exception_message", onSingleton = true, required = 1)
     public abstract static class ExceptionMessageNode extends CoreMethodArrayArgumentsNode {
 
+        @Child private FromJavaStringNode fromJavaStringNode;
+
         @Specialization(limit = "getInteropCacheLimit()")
         protected Object getExceptionMessage(Object receiver,
                 @Cached TranslateInteropExceptionNode translateInteropException,
-                @CachedLibrary("receiver") InteropLibrary receivers) {
+                @CachedLibrary("receiver") InteropLibrary receivers,
+                @Cached ConditionProfile isJavaStringProfile) {
             try {
-                return receivers.getExceptionMessage(receiver);
+                var message = receivers.getExceptionMessage(receiver);
+
+                if (isJavaStringProfile.profile(message instanceof String)) {
+                    return fromJavaString((String) message);
+                } else {
+                    return message;
+                }
```

Comment:
This should not be necessary, a j.l.String is just "a foreign String" i.e. InteropLibrary#isString.
And foreign strings support `.dup`, etc.
So we shouldn't convert eagerly here.

The problem seems to be in `IsCopyableObjectNode`:
```
jt -q ruby -e 's=Truffle::Interop.to_java_string "abc"; p s.dup'
-e:1:in `<main>': TruffleRuby doesn't have a case for the org.truffleruby.language.objects.IsCopyableObjectNodeGen node with values of type java.lang.String (TypeError)
	from org.truffleruby.language.objects.IsCopyableObjectNodeGen.executeAndSpecialize(IsCopyableObjectNodeGen.java:124)
	from org.truffleruby.language.objects.IsCopyableObjectNodeGen.execute(IsCopyableObjectNodeGen.java:61)
	from org.truffleruby.core.kernel.KernelNodes$DupNode.dup(KernelNodes.java:644)
```
Same for `jt ruby -e 's=Truffle::Interop.as_truffle_string "abc"; p s.dup'`

Reply:
I looked a it, we need to deal with `foreign_string.dup` higher than in IsCopyableObjectNode or Kernel#dup.
I think it makes most sense to also "delegate" `.dup` like other string methods, through `src/main/ruby/truffleruby/core/truffle/polyglot_methods.rb`

Reply:
I pushed a commit to do that, so the change above is no longer needed.

Reply:
Done.